rustls-libssl/src/entry.rs

```diff
@@ -1840,6 +1847,27 @@ impl Castable for SSL_CONF_CTX {
     type RustType = NotThreadSafe<conf::SslConfigCtx>;
 }
 
+entry! {
+    pub fn _X509_check_private_key(cert: *mut X509, pkey: *mut EVP_PKEY) -> c_int {
+        if cert.is_null() || pkey.is_null() {
+            return 0;
+        }
+
+        let chain = vec![CertificateDer::from(
+            OwnedX509::new_incref(cert).der_bytes(),
+        )];
+        let Ok(certified_key) = OpenSslCertifiedKey::new(chain, EvpPkey::new_incref(pkey)) else {
```

Comment:
May be worth seeing what it takes to implement the new public key function here? https://github.com/rustls/rustls-openssl-compat/blob/c11445108bc643c016841d6c3941682614d73ea1/rustls-libssl/src/sign.rs#L158